include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -546,6 +546,21 @@ make_transform_iterator(_Iter __it, _UnaryFunc __unary_func)
     return transform_iterator<_Iter, _UnaryFunc>(__it, __unary_func);
 }
 
+//functor concept
+struct __functor_concept
+{
+    template <typename _T>
+    static auto
+    test(int) -> decltype(::std::declval<_T>()(0), ::std::true_type{});
```

Comment:
int is a kind of dummy parameter, to allow a compiler differ two members with the same names. This is one of "typical" SFINAE approach to compile-time test there is some method or is not.
we need to check some callable type which has operator()(difference_type) at least.
std::is_function "checks whether T is a function type . Types like std::function, lambdas, classes with overloaded operator() and pointers to functions don't count as function types".

Reply:
Thanks for sharing the details. We might want to add a comment since this part is tricky. Also we might want to reconsider the naming: Is `__is_functor` too generic?